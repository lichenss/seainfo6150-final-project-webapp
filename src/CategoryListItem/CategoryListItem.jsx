import React from 'react'
import { Link } from 'react-router-dom';
import styles from './CategoryListItem.module.css'

const CategoryListItem = (props) => {
    return (
        <li className={styles.category}>
            <Link to={`/category/${props.category.categoryLink}`}>
            <div className={styles.imageBlock}>
                    <img
                        className={styles.image}
                        src={props.category.imageURI}
                        alt={`${props.category.categoryName}`}
                    />
                </div>
                <h3 className={styles.title}>{props.category.categoryName}</h3>
            </Link>
        </li>
    )
}
export default CategoryListItem