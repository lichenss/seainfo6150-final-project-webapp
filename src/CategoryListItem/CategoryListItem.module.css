.category {
  width: 80%;
}

li {
  list-style: none;
}

.imageBlock {
  width: 100%;
  height: 8rem;
}

.image {
  width: 100%;
  height: 100%;
}

.title {
  font-family: 'Open Sans', sans-serif;
  font-size: 1.5rem;
  margin-top: 8px;
  color: black;
}

@media screen and (min-width:600px){
  .imageBlock {
    height: 4rem;
  }
}
@media screen and (min-width:900px){
  .imageBlock {
    height: 6rem;
  }
}
@media screen and (min-width:1200px){
  .imageBlock {
    height: 8rem;
  }
}
