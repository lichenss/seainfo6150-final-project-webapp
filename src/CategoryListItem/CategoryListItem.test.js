import React from "react";
import CategoryListItem from "./CategoryListItem";
import mathandlogicImg from '../images/math_and_logic.jpg';
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
const category={
    slug: 5,
        categoryName: 'Math and Logic',
        categoryLink:'math-and-logic',
        imageURI: mathandlogicImg,
}
describe("CategoryListItem tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <CategoryListItem category={category} key={category.slug} />
      </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });