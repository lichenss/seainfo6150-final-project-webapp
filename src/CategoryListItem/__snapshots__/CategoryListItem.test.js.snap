// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItem tests renders correctly 1`] = `
<div>
  <li
    class="category"
  >
    <a
      href="/category/math-and-logic"
    >
      <div
        class="imageBlock"
      >
        <img
          alt="Math and Logic"
          class="image"
          src="math_and_logic.jpg"
        />
      </div>
      <h3
        class="title"
      >
        Math and Logic
      </h3>
    </a>
  </li>
</div>
`;
